keypair/keypairtest/slice.go

```
@@ -0,0 +1,19 @@
+package keypairtest
+
+import (
+	"github.com/stellar/go/keypair"
+)
+
+// SliceSource is a keypair.Generator source that has the values returned
+// from a slice of keys that are provided at generation one at a time.
+type SliceSource []*keypair.Full
+
+// Generate returns the first key in the slice, and then shortens the slice
+// removing the returned key, so that each call returns the next key in the
+// original source. If called when no keys are available the function will
+// panic.
```

Given that this is a library, maybe we shouldn't use panic? What do you think about returning an error if the slice is empty?

---

I think it is okay to panic in this function. This is in the `keypairtest` package and it's only designed for use in tests. I could raise errors here like "no keys in slice source" but any test where this is used should probably fail hard anyway if the developer hasn't configured the slice source with enough keys. It'll be indicative of test setup failure and not any application failure.

---

Ah my bad. I didn't realize this is defined in `keypairtest`, not `keypair`.